fili-core/src/main/java/com/yahoo/bard/webservice/data/config/provider/PhysicalTableConfiguration.java

```diff
@@ -0,0 +1,31 @@
+// Copyright 2016 Yahoo Inc.
+// Licensed under the terms of the Apache license. Please see LICENSE.md file distributed with this work for terms.
+package com.yahoo.bard.webservice.data.config.provider;
+
+import com.yahoo.bard.webservice.data.config.dimension.DimensionConfig;
+import com.yahoo.bard.webservice.data.config.names.FieldName;
+import com.yahoo.bard.webservice.data.config.table.PhysicalTableDefinition;
+
+import java.util.Set;
+
+/**
+ * A PhysicalTableConfiguration can create its PhysicalTableDefinition, given a dimension configuration.
+ */
+public interface PhysicalTableConfiguration {
+
```

Comment:
Yeah, just not in this PR.

The implementation I have for the YAML config provider: 
* Takes metric names, dimension names, and time grain in the constructor. 

* In buildPhysicalTable, you get a map of actual constructed dimension config objects. So the `dimensionMap` is used to resolve dimension names to DimensionConfig objects so a `PhysicalTableDefinition` can be built.

```
      @Override
      public PhysicalTableDefinition buildPhysicalTable(ConfigurationDictionary<DimensionConfig> dimensionMap) {
 
          Set<DimensionConfig> dimSet = dimensions
                  .stream()
                  .filter(dimensionMap::containsKey)
                  .map(dimensionMap::get)
                  .collect(Collectors.toSet());
 
          return new PhysicalTableDefinition(name, grain, dimSet);
      }
```